lbrynet/core/Session.py

```diff
@@ -241,6 +242,11 @@ def threaded_try_upnp():
 
         def upnp_failed(err):
             log.warning("UPnP failed. Reason: %s", err.getErrorMessage())
+            with TXRequestsSession() as session:
+                r = session.get('https://api.ipify.org', format='json')
+                log.info("got external ip from ipify.org")
+                self.external_ip = r.json()['ip']
+                log.info("Set external ip: %s", self.external_ip)
             return False
```

Comment:
Does it make sense to do this? If UPNP failed, what good is setting an external ip that you can't listen to?